sympy/ntheory/tests/test_generate.py

```diff
@@ -129,6 +129,9 @@ def test_generate():
     assert list(sieve.primerange(5, 9)) == [5, 7]
     sieve._reset(prime=True)
     assert list(sieve.primerange(2, 12)) == [2, 3, 5, 7, 11]
+    assert list(sieve.primerange(8)) == [2, 3, 5, 7]
+    assert list(sieve.primerange(-2)) == []
+    assert list(sieve.primerange(30)) == [2, 3, 5, 7, 11, 13, 17, 19, 23, 29]
```

Comment:
The "endpoint is not included" fact is also not tested at all. We should add tests for something like `primerange(4, 11)` and `primerange(11)`.